nomad/state/state_store.go

```diff
@@ -2025,6 +2025,14 @@ func (s *StateStore) DeleteJobTxn(index uint64, namespace, jobID string, txn Txn
 	if _, err = txn.DeleteAll("scaling_event", "id", namespace, jobID); err != nil {
 		return fmt.Errorf("deleting job scaling events failed: %v", err)
 	}
+
+	// Delete task group volume claims
+	for _, tg := range job.TaskGroups {
+		if _, err = txn.DeleteAll(TableTaskGroupVolumeClaim, indexID, namespace, jobID, tg.Name); err != nil {
```

Comment:
This would become `txn.DeletePrefix` if we go with the schema change I recommend in `feasibility.go`